sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/sql/FlinkSqlGenerator.java

```diff
@@ -0,0 +1,453 @@
+      SqlCall call = (SqlCall) watermarkExpression.get();
+
+      SqlNode expr = stripAs(call);
+
+      if (expr instanceof SqlIdentifier && ReservedName.SOURCE_TIME.matches(((SqlIdentifier)expr).getSimple())) {
+        waterMarkType = WaterMarkType.SOURCE_WATERMARK;
+      } else {
+        waterMarkType = WaterMarkType.COLUMN_BY_NAME;
+      }
+    }
+
+    String name = getWatermarkName(watermarkColumn, watermarkExpression);
+
+    if (!setMetadata && waterMarkType == WaterMarkType.SOURCE_WATERMARK) {
+      return new SqlWatermark(SqlParserPos.ZERO,
+          identifier(name),
+          lightweightOp("SOURCE_WATERMARK").createCall(SqlParserPos.ZERO));
+    } else {
+      com.google.common.base.Preconditions.checkArgument(
+          setMetadata || waterMarkType == WaterMarkType.COLUMN_BY_NAME);
+      return new SqlWatermark(SqlParserPos.ZERO,
+          identifier(name),
+          boundedStrategy(name, "1"));
+    }
+
+  }
+
+  private String getWatermarkName(Optional<SqlNode> watermarkColumn,
+      Optional<SqlNode> watermarkExpression) {
+    if (watermarkColumn.isPresent()) { //watermark is a timestamp column
+      return removeAllQuotes(RelToFlinkSql.convertToString(watermarkColumn.get()));
+    } else { //watermark is a timestamp expression
+      Preconditions.checkArgument(watermarkExpression.isPresent());
+      SqlCall call = (SqlCall) watermarkExpression.get();
+      SqlNode name = call.operand(1);
+      return removeAllQuotes(RelToFlinkSql.convertToString(name));
+    }
+  }
+
+  private SqlNodeList createColumns(Optional<Schema> schemaOpt, RelDataType relDataType,
+      Optional<SqlNode> watermarkExpression, Optional<SqlNode> watermarkColumn,
+      Optional<String> sourceTimeMetaData, boolean setMetadata) {
+    if (relDataType.getFieldList().isEmpty()) {
+      return SqlNodeList.EMPTY;
+    }
+    return createColumns(relDataType.getFieldList(),
+        watermarkColumn, watermarkExpression, setMetadata, sourceTimeMetaData);
+
+  }
+
+  public SqlNodeList createColumns(List<RelDataTypeField> fields,
+      Optional<SqlNode> watermarkColumn,
+      Optional<SqlNode> watermarkExpression,
+      boolean setMetadata,
+      Optional<String> sourceTime
+  ) {
+
+    List<SqlNode> nodes = new ArrayList<>();
+
+    //TODO: This is brittle, it mirrors the structure of UniversalTable by index
+    int index = 0;
+    for (RelDataTypeField column : fields) {
+      SqlNode node;
+      if (setMetadata && index==0) {
+        node = new SqlTableColumn.SqlComputedColumn(SqlParserPos.ZERO,
+            identifier(column.getKey()), null,
+            lightweightOp(UUID_FCT_NAME).createCall(SqlParserPos.ZERO));
+      } else if (setMetadata && index==1) {
+        node = new SqlTableColumn.SqlComputedColumn(SqlParserPos.ZERO,
+            identifier(column.getKey()), null,
+            lightweightOp("PROCTIME").createCall(SqlParserPos.ZERO));
+      } else if (setMetadata && index==2 && sourceTime.isPresent()) {
```

Comment:
Likewise, metadata is something that should be configured in the table.json as specified in this document:
https://docs.google.com/document/d/1RtcsG7A7fU55xlz6VzbwCTajyuPVVzOssiwfe5Ju9hA/edit?pli=1#heading=h.4zhnhh3lprav